src/api/ledger/parse/pathfind.js

```diff
@@ -8,17 +8,25 @@ function parsePaths(paths) {
     _.omit(step, ['type', 'type_hex'])));
 }
 
+function ensureIOUCounterparty(amount, sourceAmount) {
+  const defaults = sourceAmount.counterparty ?
+    {counterparty: sourceAmount.counterparty} : {};
+  return amount.currency === 'XRP' || amount.counterparty ? amount :
+    _.assign(defaults, amount);
+}
+
 function parsePathfind(sourceAddress: string,
     destinationAmount: Object, pathfindResult: Object): Object {
   return pathfindResult.alternatives.map(function(alternative) {
+    const sourceAmount = parseAmount(alternative.source_amount);
     return {
       source: {
         address: sourceAddress,
-        amount: parseAmount(alternative.source_amount)
+        amount: sourceAmount
       },
       destination: {
         address: pathfindResult.destination_account,
-        amount: destinationAmount
+        amount: ensureIOUCounterparty(destinationAmount, sourceAmount)
```

Comment:
lol, I guess strictly speaking, paths are built with Account/Destination too, but I think you got what I was trying to convey!